portfolio/src/main/webapp/jquery_functions.js

```diff
@@ -1,31 +1,59 @@
 /*
 * Purpose: contains various JQuery functions that listen for events on main portfolio page
 */
+// For each individual background img, width is 2016px height is 1512px
+const BACKGROUND_IMG_WIDTH = 2016;
+const BACKGROUND_IMG_HEIGHT = 1512;
+const VERT_PADDING = 40;
+const HORZ_PADDING = 20;
+const SPRITE_IMG_ROWS = 2;
+const SPRITE_IMG_COLS = 3;
+const NUM_IMAGES = 5;
 
-let img_arr = new Array(5);
-img_arr = ["../images/mountbierstadt.jpg", 
-           "../images/mountyale.jpg",
-           "../images/gardenofgods.jpg",
-           "../images/lakedillon.jpg",
-           "../images/mountelbert.jpg"
-           ];
 labels = ["the summit of Mount Bierdstadt.", 
           "the summit of Mount Yale.", 
+          "the summit of Mount Elbert, the tallest point in Colorado.",
           "the Garden of the Gods.", 
-          "Lake Dillon, seen from the vantage point of driving along Dillon Dam road.",
-          "the summit of Mount Elbert, the tallest point in Colorado."];
+          "Lake Dillon, seen from the vantage point of driving along Dillon Dam road."];
 
+profile_img_paths = ["images/noogler.gif",
+                     "images/zoebakercircle.png"
+                    ];
+
+function preload_imgs(img_arr, img_paths){
+  for (let i = 0; i < img_paths.length; i++) {
+      img_arr[i] = new Image();
```

Comment:
So you can make a new variable for the image object, `temp_image = new Image(); temp_image.src = img_paths[]; img_array[i] = temp_image;`

Additionally, what is the case where `img_paths.length > img_arr.length`?

Reply:
I don't quite understand why I would need to explicitly make a new variable with each loop, would you be able to explain that to me?

And good point about the additional case, changed it so that I just allocate the correct size image array in the function with the length of img_paths.

Reply:
You don't need it explicitly make one, it's just good practice to delineate between creating and assigning. `img_arr[i] = new Image();` you are doing both, which is fine. It's just a preference for reading

Reply:
Gotcha, makes sense.